server: make engine stall checker warn, not fatal, by default

tbg opened the pull request:
The environment variable will allows us to go back to testing the
absence of stalls, though we may also chose to do something more
programmatic based on RocksDB's internal tracking of write stalls.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/36484)
<!-- Reviewable:end -->

tbg committed:
server: make engine stall checker warn, not fatal, by default

The environment variable will allows us to go back to testing the
absence of stalls, though we may also chose to do something more
programmatic based on RocksDB's internal tracking of write stalls.

Release note: None

bdarnell reviewed (approved):
We should probably make this a cluster setting instead of env vars. Originally we didn't think anyone was going to have to touch it, but we were wrong about that. I think we're going to need to recommend making stalls fatal for EBS users (or maybe not - we haven't seen the EBS stalls in a while AFAIK). 

We should also get telemetry when this is reached. 

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/36484#-:-LbZE_BvCKJkOfqhakJC:b-x7p1r0)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tbg commented:
How strongly do you feel about the cluster setting? Right now we can't recommend that anyone turn it on due to the misfiring we've seen, EBS or not. https://github.com/cockroachdb/cockroach/pull/36408 will give us RocksDB logging, I don't know if that will make a stall easier to see, I'd hope so (maybe @ajkr knows off-hand).

ajkr commented:
Yes, we are soon going to be painfully aware of every time RocksDB goes into stall mode, or even slowdown mode.

bdarnell commented:
A few months ago we saw a rash of stalls on EBS volumes that motivated the introduction of this feature. Back then I thought it would be important to get this enabled. But I don't think we've seen any more occurrences since then, so maybe it's less important now. I think I'm OK with leaving it in env vars for now.

tbg commented:
> Yes, we are soon going to be painfully aware of every time RocksDB goes into stall mode, or even slowdown mode.

Just to be clear, this isn't primarily about RocksDB write stalls (which is what this feature has been finding most of the time) but about disk I/O stalls. If writing to disk just started blocking indefinitely, would the RocksDB logging give us anything good (not asking you to go find out if you don't know).

tbg commented:
TFTR!

bors r=bdarnell

ajkr commented:
> Just to be clear, this isn't primarily about RocksDB write stalls (which is what this feature has been finding most of the time) but about disk I/O stalls. If writing to disk just started blocking indefinitely, would the RocksDB logging give us anything good (not asking you to go find out if you don't know).

Got it. I can't think of any rocks warnings that would help us here.

priitp commented:
Will it be backported to 19.1? We're hitting this feature on physical hw.

tbg commented:
Yes, absolutely.
